File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        int count[201]={0};
        
        int m = 0;
        for(int n:nums) {
            count[n]++;
            m = max(m, count[n]);
        }
        vector<vector<int>> ans;
        for(int i=0;i<m;i++)
            ans.push_back(vector<int>());
        for(int i=1;i<=200;i++) {
            for(int j=0;j<count[i];j++)
                ans[j].push_back(i);
        }
        return ans;
    }
};

File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.java
class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        int[] count = new int[201];
        int max = 0;
        for(int n:nums) {
            count[n]++;
            max = Math.max(max, count[n]);
        }
        List<List<Integer>> ans = new ArrayList<>();
        for(int i=0;i<max;i++)
            ans.add(new ArrayList<>());
        for(int i=1;i<=200;i++) {
            for(int j=0;j<count[i];j++)
                ans.get(j).add(i);
        }
        return ans;
    }
}